docs/src/main/asciidoc/cache.adoc

```diff
@@ -323,7 +320,22 @@ method annotated with `@CacheResult` or `@CacheInvalidate`.
 
 This annotation is optional and should only be used when some of the method arguments are NOT part of the cache key.
 
-=== Composite cache key building logic
+[#cache-keys-building-logic]
+=== Cache keys building logic
+
+Cache keys are built by the annotations API using the following logic:
+
+* If an `io.quarkus.cache.CacheKeyGenerator` is declared in a `@CacheResult` or a `@CacheInvalidate` annotation, then it is used to generate the cache key. The `@CacheKey` annotations that might be present on some of the method arguments are ignored.
+* Otherwise, if the method has no arguments, then the cache key is an instance of `io.quarkus.cache.DefaultCacheKey` built from the cache name.
+* Otherwise, if the method has exactly one argument, then that argument is the cache key.
+* Otherwise, if the method has multiple arguments but only one annotated with `@CacheKey`, then that annotated argument is the cache key.
+* Otherwise, if the method has multiple arguments annotated with `@CacheKey`, then the cache key is an instance of `io.quarkus.cache.CompositeCacheKey` built from these annotated arguments.
+* Otherwise, the cache key is an instance of `io.quarkus.cache.CompositeCacheKey` built from all the method arguments.
```

Comment:
It's a little hard to follow all these `otherwise` statements, so maybe the last one should mention the conditions it applies to?

Reply:
Do you think starting with `Otherwise, if the method has multiple arguments and none of them are annotated with @CacheKey` would be better? From a code perspective, that last sentence is a simple `else` block.

Reply:
> would be better?

I do.

> From a code perspective, that last sentence is a simple else block

Yeah I understand that, I am just saying that it's very hard to follow that logic in english writing :)